javaslang/src/main/java/javaslang/collection/List.java

```diff
@@ -1390,12 +1396,12 @@ default String stringPrefix() {
 
     @Override
     default List<T> take(int n) {
-        if (n >= length()) {
-            return this;
-        }
         if (n <= 0) {
             return empty();
         }
+        if (n >= length()) {
```

Comment:
I don't know, but in every case it's a microoptimizatio, I suggest we only change it if it's easier to understand, I really doubt you could measure any speed difference :)